lhotse/cut.py

```diff
@@ -1531,9 +1535,84 @@ def cut_into_windows(self, duration: Seconds, keep_excessive_supervisions: bool
+        :param duration: an optional float in seconds.
+            When ``None``, we will preserve the duration of the cuts in ``self``
+            (i.e. we'll truncate the mix if it exceeded the original duration).
+            Otherwise, we will keep sampling cuts to mix in until we reach the specified
+            ``duration`` (and truncate to that value, should it be exceeded).
+        :param snr: an optional float, or pair (range) of floats, in decibels.
+            When it's a single float, we will mix all cuts with this SNR level
+            (where cuts in ``self`` are treated as signals, and cuts in ``cuts`` are treated as noise).
+            When it's a pair of floats, we will uniformly sample SNR values from that range.
+            When ``None``, we will mix the cuts without any level adjustment
+            (could be too noisy for data augmentation).
+        :param mix_prob: an optional float in range [0, 1].
+            Specifies the probability of performing a mix.
+            Values lower than 1.0 mean that some cuts in the output will be unchanged.
+        :return: a new ``CutSet`` with mixed cuts.
+        """
+        assert 0.0 <= mix_prob <= 1.0
+        assert duration is None or duration > 0
+        mixed_cuts = []
+        for cut in self:
+            # Check whether we're going to mix something into the current cut
+            # or pass it through unchanged.
+            if random.uniform(0.0, 1.0) > mix_prob:
+                mixed_cuts.append(cut)
+                continue
+            # Randomly sample a new cut from "cuts" to mix in.
+            to_mix = cuts.sample()
+            # Determine the SNR - either it's specified or we need to sample one.
+            snr = snr if isinstance(snr, (float, type(None))) else random.uniform(*snr)
+            # Actual mixing
+            mixed = cut.mix(other=to_mix, snr=snr)
+            # Did the user specify a duration?
+            # If yes, we will ensure that shorter cuts have more noise mixed in
+            # to "pad" them with at the end.
+            if duration is not None:
+                mixed_in_duration = to_mix.duration
+                # Keep sampling until we mixed in a "duration" amount of noise.
+                while mixed.duration < duration:
+                    to_mix = cuts.sample()
+                    # Keep the SNR constant for each cut from "self".
+                    mixed = mixed.mix(other=to_mix, snr=snr, offset_other_by=mixed_in_duration)
+                    # Since we're adding floats, we can be off by an epsilon and trigger
+                    # some assertions for exceeding duration; do precautionary rounding here.
+                    mixed_in_duration = round(mixed_in_duration + to_mix.duration, ndigits=8)
+            # We truncate the mixed to either the original duration or the requested duration.
+            mixed = mixed.truncate(duration=cut.duration if duration is not None else duration)
```

Comment:
should be 
```python
if duration is None
```

Reply:
Thanks, I'll add tests before merging ;)